### PR TITLE
KEYCLOAK-6750 KEYCLOAK-9597 Adapt Tomcat adapter tests to new structure

### PR DESCRIPTION
**Pull request by mhajas**
Testing DemoServletAdapterTest and SessionServletAdapterTest

### CODE REVIEW

File: testsuite/integration-arquillian/servers/app-server/tomcat/tomcat9/pom.xml
Comment (vramik) [inline testsuite/integration-arquillian/servers/app-server/tomcat/tomcat9/pom.xml]: We should use keycloak-tomcat9-adapter-dist and keycloak-saml-tomcat9-adapter-dist which are not yet done. 

@hmlnarik can you suggest a way how to proceed with this one? I can see two possibilities we either add the adapter for tomcat9 within this PR or we don't add the tomcat9 test modules and wait for the adapters which should be done outside of this. (Or we can update this PR to use tomcat9 adapters and just don't include tests to the pipeline till the adapters are done) WDYT?

File: testsuite/integration-arquillian/servers/app-server/tomcat/tomcat7/pom.xml
Comment (vramik) [inline testsuite/integration-arquillian/servers/app-server/tomcat/tomcat7/pom.xml]: I'd suggest `App Server - Tomcat X` as a name, it'd be consistent with other app-server modules (the same for both Tomcat 8 and Tomcat 9)

**Review by vramik — CHANGES_REQUESTED** [review]
Thank you @mhajas for the work done so far. I have some comments: 

In testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml there is the <[execution](https://github.com/keycloak/keycloak/pull/5880/files#diff-3d89669198b288a2e955fc1fcc1853c1R110)> which is imho not required any more, would you mind to test it, pls? 

The same applies to the <[plugin](https://github.com/keycloak/keycloak/pull/5880/files#diff-3d89669198b288a2e955fc1fcc1853c1R168)>, can you test if it is required?

What about keycloak/testsuite/tomcat7 and keycloak/testsuite/tomcat8 was those migrated? If so shouldn't those be also deleted (and updated travis script)?

File: testsuite/integration-arquillian/servers/app-server/tomcat/tomcat9/pom.xml
Comment (mhajas) [reply testsuite/integration-arquillian/servers/app-server/tomcat/tomcat9/pom.xml]: It looks like this artifact contains only some common classes for all versions of tomcat containers. However, there is no managed container for Tomcat 9 yet. I used the container for Tomcat 8, but we should upgrade for Tomcat 9 once it is released.

**Comment by vramik** [comment]
@mhajas lgtm now, let me know when the PR is ready for final review, thanks

File: testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java
Comment (slaskawi) [inline testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java]: Probably it would be useful to either remove the empty javadoc (the same for the rest of files in this PR) or put something meaningful here. But It's a nitpick.

File: testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml
Comment (slaskawi) [inline testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml]: I'm not sure, but it seems you can also skip building the `common` module when all Tomcat profiles are disabled. In Jetty adapter PR, I simply added `common` module to every profile and skipped it with default execution.

File: testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml
Comment (slaskawi) [inline testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml]: Could we extract the versions to parent pom (by using some BOM or by declaring versions in properties). This should make it a bit easier during the upgrades.

The same comment applies to other dependencies in other files.

File: testsuite/integration-arquillian/servers/app-server/tomcat/tomcat9/pom.xml
Comment (slaskawi) [inline testsuite/integration-arquillian/servers/app-server/tomcat/tomcat9/pom.xml]: Jar packaging is the default one and can be removed.

File: testsuite/integration-arquillian/tests/base/pom.xml
Comment (slaskawi) [inline testsuite/integration-arquillian/tests/base/pom.xml]: If you don't want to implement it in this PR, just throw an `UnsupportedOperationException` when you discover that it's not 0.

**Review by slaskawi — CHANGES_REQUESTED** [review]
This is really great work @mhajas! I really like it! 

I only found a few, really minor things to fix. Otherwise looks good to me.

File: testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java
Comment (mhajas) [reply testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java]: There is @author in this comment. Plenty of files have such comment on top of it. I think it is good for others to see who to contact in case of some questions. But I can remove it.

File: testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml
Comment (mhajas) [reply testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml]: I am not sure we want to extract those. These are just dependencies for tomcat, we are not using it anywhere in testsuite except for this place. Even if we want to use those dependencies we may need some other version.

File: testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml
Comment (mhajas) [reply testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml]: Good idea. Thanks

**Comment by vramik** [comment]
@keycloak-ci-bot test

**Comment by mhajas** [comment]
After merge of https://github.com/keycloak/keycloak/pull/5885 I think we can also remove this package
https://github.com/keycloak/keycloak/tree/master/testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/helper/adapter

**Comment by mhajas** [comment]
@vramik @slaskawi I think this is ready for final review

File: testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java
Comment (slaskawi) [reply testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java]: Sure, no problem.

As a side note - I always use `git gui blame` or `git annotate` to find the author. Javadocs like that often are a bit misleading (of course in my opinion), since the file changes over the time.

Bu anyway - granted. Please keep it or remove it if you'd like.

File: testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml
Comment (slaskawi) [reply testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml]: Fair point. Is there any BOM available that would include them? I'm just a little bit concerned how do upgrade those artifacts in the future and make sure they work together? If there's no other option, please keep it as it.

**Review by slaskawi — DISMISSED** [review]
LGTM. I added two comments but I leave them up to @mhajas. You may leave them as they are or implement modifications. It's your call.

File: testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java
Comment (mhajas) [reply testsuite/integration-arquillian/servers/app-server/tomcat/common/src/main/java/org/keycloak/testsuite/arquillian/tomcat/container/CommonTomcatDeploymentArchiveProcessor.java]: You are right. Removed. Thanks

File: testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml
Comment (mhajas) [reply testsuite/integration-arquillian/servers/app-server/tomcat/pom.xml]: I am not sure, I didn't find anything. I used version from keycloak/pom.xml where it is possible and extracted others to tomcat parent pom properties. Are you ok with this solution?

**Comment by hmlnarik** [comment]
@keycloak-ci-bot test

**Comment by mhajas** [comment]
@keycloak-ci-bot test

**Review by hmlnarik — APPROVED** [review]

**Review by vramik — APPROVED** [review]

**Pull request merged**
